service/src/parentchain_handler.rs

```diff
@@ -65,12 +67,12 @@ pub(crate) struct ParentchainHandler<ParentchainApi, EnclaveApi> {
 
 // #TODO: #1451: Reintroduce `ParentchainApi: ChainApi` once there is no trait bound conflict
 // any more with the api-clients own trait definitions.
-impl<EnclaveApi> ParentchainHandler<ParentchainApi, EnclaveApi>
+impl<EnclaveApi> ParentchainHandler<IntegriteeParentchainApi, EnclaveApi>
```

Comment:
You are exactly right. I believe this is the root cause of the problem. The api-client has re-defined types and traits because they for not available in no-std in substrate < v0.9.43. When I updated the api-client to its newest version, but v0.9.42, I had to reintroduce these types again: https://github.com/scs/substrate-api-client/commit/ea49a0d11f83f15dd51bcbb556d35c3a13780c63.

Then I was essentially in the same rabbit whole as you: certain trait-bounds weren't satisfied because we are using the actual types from substrate whereas the api-client uses its types/traits in its traitbounds.

My recommedation is: if you have to use generics in those api taits, we unfortunately have to upgrade the polkadot version and the api-client.